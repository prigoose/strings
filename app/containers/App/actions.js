/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
  POST_STRING_TO_DB,
} from './constants';

/**
 * Load the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRINGS
 */
export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

/**
 * Dispatched when the strings are loaded by the request saga
 *
 * @param  {array} strings The data
 *
 * @return {object}      An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
 */
export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

export function stringLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}

/**
 * Load the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of INSERT_STRING
 */
export function postStringToDb() {
  return {
    type: POST_STRING_TO_DB,
  };
}
