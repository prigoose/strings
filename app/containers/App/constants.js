/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const LOAD_STRINGS = 'strings/App/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'strings/App/LOAD_STRINGS_SUCCESS';
export const LOAD_STRINGS_ERROR = 'strings/App/LOAD_STRINGS_ERROR';
export const INSERT_STRING = 'strings/App/INSERT_STRING';
export const INSERT_STRING_SUCCESS = 'strings/App/INSERT_STRING_SUCCESS';
export const INSERT_STRING_ERROR = 'strings/App/INSERT_STRING_ERROR';
export const UPDATE_VALUE = 'strings/App/UPDATE_VALUE';
