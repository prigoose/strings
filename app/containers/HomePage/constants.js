/*
 *
 * HomePage constants
 *
 */

export const GET_STRINGS = 'app/HomePage/GET_STRINGS';
