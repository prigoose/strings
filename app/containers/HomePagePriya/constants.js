/*
 *
 * HomePagePriya constants
 *
 */

export const GET_STRINGS = 'app/HomePagePriya/GET_STRINGS';
