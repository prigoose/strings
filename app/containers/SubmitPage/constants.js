/*
 *
 * SubmitPage constants
 *
 */

export const CHANGE_STRING = 'app/SubmitPage/CHANGE_STRING';
