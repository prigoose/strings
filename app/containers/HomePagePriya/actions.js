/*
 *
 * HomePagePriya actions
 *
 */

import { GET_STRINGS } from './constants';

export function getStrings() {
  return {
    type: GET_STRINGS,
  };
}
