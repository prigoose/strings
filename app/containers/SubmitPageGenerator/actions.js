/*
 *
 * SubmitPageGenerator actions
 *
 */

import { CHANGE_STRING } from './constants';

export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}
