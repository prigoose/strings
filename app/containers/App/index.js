/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

// import HomePage from 'containers/HomePage/Loadable';
// import SubmitPage from 'containers/SubmitPage/Loadable';
import HomePagePriya from 'containers/HomePagePriya/Loadable';
import SubmitPageGenerator from 'containers/SubmitPageGenerator';

import Header from 'components/Header';
import H1 from 'components/H1';

const AppWrapper = styled.div`
  max-width: calc(768px + 16px * 2);
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  padding: 0 16px;
  flex-direction: column;
`;

export default function App() {
  return (
    <div>
      <AppWrapper>
        <Header />
        <H1>Dovenmuehle String Extravaganza</H1>
        <Switch>
          <Route exact path="/" component={HomePagePriya} />
          <Route exact path="/submit" component={SubmitPageGenerator} />
        </Switch>
      </AppWrapper>
    </div>
  );
}
