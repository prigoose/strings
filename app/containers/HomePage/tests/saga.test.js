import { call, put, takeLatest } from 'redux-saga/effects';
import { LOAD_STRINGS } from 'containers/App/constants';
import { stringsLoaded, stringLoadingError } from 'containers/App/actions';
import request from 'utils/request';

import stringData, { getStrings, requestURL } from '../saga';


const strings = [
  {
    string: 'HelloKitty',
  },
  {
    string: 'Priya Ghose',
  },
];

describe('successful call to getStrings Saga', () => {
  const getStringsGenerator = getStrings();
  it('should make a request', () => {
    expect(getStringsGenerator.next().value).toEqual(call(request, requestURL));
  });
  it('should dispatch action once strings load', () => {
    expect(getStringsGenerator.next(strings).value).toEqual(put(stringsLoaded(strings)));
  });
  it('should be finished', () => {
    expect(getStringsGenerator.next().done).toEqual(true);
  });
});

// TO DO: research how to appropriately test a failed call.
// describe('failed call to getStrings Saga', () => {
//   const getStringsGenerator2 = getStrings();
//   it('should not match a request to a bad URL', () => {
//     expect(getStringsGenerator2.next().value).toNotEqual(call(request, 'http://xyz.com'));
//   });
//   const error = call(request, 'http://xyz.com');
//   it('should error', () => {
//     expect(getStringsGenerator2.next().next(error)).toEqual(put(stringLoadingError(error)));
//   });
//   it('should be finished', () => {
//     expect(getStringsGenerator2.next().done).toEqual(true);
//   });
// });
