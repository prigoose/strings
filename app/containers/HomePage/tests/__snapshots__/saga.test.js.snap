// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrings Saga should dispatch the stringsLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3000/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getStrings Saga should dispatch the stringsLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "strings": Array [
        Object {
          "string": "HelloKitty",
        },
        Object {
          "string": "Priya Ghose",
        },
      ],
      "type": "strings/App/LOAD_STRINGS_SUCCESS",
    },
    "channel": null,
  },
}
`;
